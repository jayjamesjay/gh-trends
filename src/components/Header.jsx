import React from 'react';
import PropTypes from 'prop-types';

import { MenuToggle } from './Nav';
import StyledHeader from '../styles/Headers';
import { HeaderLink } from '../styles/Link';
import { Img } from '../styles/Img';
import { ButtonTheme } from '../styles/Button';
import SwitchThemeImg from '../assets/img/dark-mode.svg';

/**
 * @module Header
 */

/**
 * Header of the website
 *
 * @property {string} link - URL pointing to Home Page
 * @property {string} title - displayed text
 * @property {function} switchTheme - switches theme (dark/light)
 * @property {boolean} hide - is nav hidden
 * @property {function} toggle - function called when MenuToggle is clicked
 * @returns {Header}
 */
export default function Header({ link, title, switchTheme, $hide, toggle }) {
  return (
    <StyledHeader>
      <HeaderLink to={link}>{title}</HeaderLink>
      <ButtonTheme onClick={switchTheme}>
        <Img src={SwitchThemeImg} alt="Toggle dark/light mode" />
      </ButtonTheme>
      <MenuToggle toggle={toggle} $open={!$hide} />
    </StyledHeader>
  );
}

Header.propTypes = {
  $hide: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  switchTheme: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
};
