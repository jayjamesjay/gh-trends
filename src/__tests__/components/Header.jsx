import React from 'react';
import 'jest-styled-components';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Header from '../../components/Header';

describe('<Header />', () => {
  const title = 'GH Trends';
  const func = () => {};

  it(`renders with default style`, () => {
    const { asFragment } = render(
      <Router>
        <Header link="" title={title} switchTheme={func} $hide={false} toggle={func} />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with title', () => {
    render(
      <Router>
        <Header link="" title={title} switchTheme={func} $hide={false} toggle={func} />
      </Router>,
    );

    expect(screen.getByRole('link').textContent).toEqual(title);
  });
});
