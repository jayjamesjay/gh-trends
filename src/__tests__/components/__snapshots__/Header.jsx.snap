// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders with default style 1`] = `
<DocumentFragment>
  .c3 {
  cursor: pointer;
  background: #fff;
  border: 0;
  border-radius: 0.5rem;
  font-size: 1.2rem;
  color: #000;
  text-transform: uppercase;
}

.c4 {
  display: flex;
  align-items: center;
  width: 3rem;
  height: 3rem;
  margin: 0 auto;
  background: transparent;
}

.c6 {
  margin: 0 1rem;
  padding: 0;
  background: transparent;
  overflow: hidden;
}

.c7 {
  display: block;
  width: 1.8rem;
  height: 0.2rem;
  margin: 0.3rem 0;
  background: #000;
  transition: transform 0.2s linear;
}

.c7:first-child {
  transform: translateY(0.45rem) rotate(-45deg);
}

.c7:nth-child(2) {
  transform: translateX(5rem);
}

.c7:last-child {
  transform: translateY(-0.45rem) rotate(45deg);
}

.c1 {
  padding: 0.5rem;
  font-size: 1.3rem;
  font-family: 'Noto Sans HK',sans-serif;
  color: #000;
  text-decoration: none;
}

.c2 {
  margin: 0.5rem;
  font-weight: 700;
  font-size: 2rem;
}

.c0 {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: env(safe-area-inset-top) env(safe-area-inset-right) 0 env(safe-area-inset-left);
  position: sticky;
  top: 0;
  z-index: 200;
  background: #fff;
  box-shadow: 0 1px 2px 0 rgba(60,64,67,0.3),0 2px 6px 2px rgba(60,64,67,0.15);
  text-transform: uppercase;
}

.c5 {
  width: 100%;
  height: auto;
  filter: invert(0%);
}

@media screen and (min-width:40rem) {
  .c4 {
    margin: 0;
  }
}

@media screen and (min-width:40rem) {
  .c6 {
    display: none;
  }
}

@media screen and (min-width:40rem) {
  .c0 {
    justify-content: flex-start;
  }
}

<header
    class="c0"
  >
    <a
      class="c1 c2"
      data-discover="true"
      font-size="1.3rem"
      href="/"
    >
      GH Trends
    </a>
    <button
      class="c3 c4"
    >
      <img
        alt="Toggle dark/light mode"
        class="c5"
        src="test-file-stub"
      />
    </button>
    <button
      aria-label="Open menu"
      class="c3 c6"
    >
      <span
        class="c7"
        open=""
      />
      <span
        class="c7"
        open=""
      />
      <span
        class="c7"
        open=""
      />
    </button>
  </header>
</DocumentFragment>
`;
